### PR TITLE
Allow Symfony 7

### PR DESCRIPTION
**Pull request by Toflar**
5.3 LTS requires Symfony 6.4 LTS as per our roadmap.

This means, Contao 5.4 must allow Symfony 7 components.

Dependencies:

- [x] `contao/manager-plugin`: https://github.com/contao/manager-plugin/pull/56
- [x] `contao/image`: https://github.com/contao/image/pull/104
- [x] `terminal42/service-annotation-bundle`: https://github.com/terminal42/service-annotation-bundle/pull/9
- [x] `terminal42/escargot`: https://github.com/terminal42/escargot/pull/36
- [x] Merge https://github.com/contao/contao/pull/7285 in `5.3`and then uptream into `5.x`

### CODE REVIEW

File: calendar-bundle/composer.json [hunk 1]
```diff
@@ -30,20 +30,20 @@
...
+        "symfony/yaml": "^6.4 || ^7.0"
     },
     "require-dev": {
         "contao/manager-plugin": "^2.3.1",
         "contao/test-case": "self.version",
         "phpunit/phpunit": "^9.5",
-        "symfony/http-client": "^6.4",
-        "symfony/phpunit-bridge": "^6.4"
+        "symfony/http-client": "^6.4 || ^7.0 || ^7.0",
+        "symfony/phpunit-bridge": "^6.4 || ^7.0 || ^7.0"
```
Comment (fritzmg) [inline calendar-bundle/composer.json]:
```suggestion
        "symfony/http-client": "^6.4 || ^7.0",
        "symfony/phpunit-bridge": "^6.4 || ^7.0"
```

File: composer.json [hunk 2]
```diff
@@ -94,70 +94,70 @@
         "psr/container": "^2.0",
         "psr/http-message": "^1.0 || ^2.0",
         "psr/log": "^1.1 || 2.0 || ^3.0",
-        "scheb/2fa-backup-code": "^6.0",
-        "scheb/2fa-bundle": "^6.0",
-        "scheb/2fa-trusted-device": "^6.0",
+        "scheb/2fa-backup-code": "^7.0",
+        "scheb/2fa-bundle": "^7.0",
+        "scheb/2fa-trusted-device": "^7.0",
```
Comment (aschempp) [inline composer.json]: Version 7 only supports PHP 8.2, but we still support PHP 8.1. There is no other functional difference for us (see https://github.com/scheb/2fa/blob/7.x/UPGRADE.md#6x-to-7x)

```suggestion
        "scheb/2fa-backup-code": "^6.0 || ^7.0",
        "scheb/2fa-bundle": "^6.0 || ^7.0",
        "scheb/2fa-trusted-device": "^6.0 || ^7.0",
```

File: core-bundle/composer.json [hunk 3]
```diff
@@ -84,69 +84,69 @@
...
         "psr/cache": "^3.0",
         "psr/container": "^2.0",
         "psr/http-message": "^1.0 || ^2.0",
         "psr/log": "^1.1 || 2.0 || ^3.0",
-        "scheb/2fa-backup-code": "^6.0",
-        "scheb/2fa-bundle": "^6.0",
-        "scheb/2fa-trusted-device": "^6.0",
+        "scheb/2fa-backup-code": "^7.0",
+        "scheb/2fa-bundle": "^7.0",
+        "scheb/2fa-trusted-device": "^7.0",
```
Comment (aschempp) [inline core-bundle/composer.json]:
```suggestion
        "scheb/2fa-backup-code": "^6.0 || ^7.0",
        "scheb/2fa-bundle": "^6.0 || ^7.0",
        "scheb/2fa-trusted-device": "^6.0 || ^7.0",
```

**Review by aschempp — CHANGES_REQUESTED** [review]

**Comment by aschempp** [comment]
see https://github.com/Toflar/contao/pull/16 for fixing the `LocaleSubscriberTest` correctly

**Comment by Toflar** [comment]
This PR is done and ready for review. What's missing is an upstream merge of 5.3 to 5.x to fix the `LocaleSubscriber` test.
After that, all packages can be updated to Symfony 7 versions except for one:

```
% composer show | grep 6.4                   
symfony/monolog-bridge                             6.4.8                 Provides integration for Monolog with various Symfony components
```

And this, we currently cannot provide because:

```
% composer why-not symfony/monolog-bridge 7.0
symfony/monolog-bridge v7.0.0 requires         monolog/monolog (^3)                     
contao/managed-edition -      does not require monolog/monolog (but 2.9.3 is installed) 
```

So it comes down to https://github.com/contao/contao/pull/7142 to also allow this very last package to be used in v7 :)

**Review by ausi — DISMISSED** [review]
❤️

**Comment by leofeyer** [comment]
#7142 has just been merged 🎉

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Pull request merged**